osutil/testhelper.go

```diff
@@ -38,3 +38,11 @@ func MustBeTestBinary(panicMsg string) {
 		panic(panicMsg)
 	}
 }
+
+// A checker for the logic that shall not run in a test context
```

Comment:
this is an internal helper so it doesn't matter too much, but goland go comments by convention start by mentioning the name of the documented construct. I recommend reading https://go.dev/doc/effective_go#commentary and effective go in general.

Reply:
Sorry, forgot about his one. Thank you for reminding. I can fix it right away.

Reply:
Sorry for being a terrible nitpicker - I personally would not (ye) create this helper, it's only used once and only has two lines so I would "unroll" it into the place where it is used. Of course once we use this function/patern many time things are different.

Reply:
@mvo5, in this very form it appears twice in our codebase. But the second time is outside the `osutil` package. 
However, the major reason behind this change was strictly document-oriented.  To notify that this code may not run during tests. So, that the next developer will be cautious before changing here anything. However, if you prefer unfolding it, I will do it.

Reply:
I agree with mvo, I think that this should just be moved to where it is called, further to pedronis' point about not encouraging developers to use it, having it as an unexported helper still leaves open the possibility that someone sees this helper and wants to use it, so they file a PR to export it and then we have this conversation again. Having this code in individual places in the codebase where the behavior is absolutely necessary reduces this risk as more developers get onboarded in the snapd codebase

Reply:
Ian, I totally understand your point and the reasoning behind it. Thank you so much for the thorough explanation. I appreciate it a lot. If I may reflect what I understand in my words - we want such sensitive pieces of code to be as plain as possible to keep them totally exposed to the eye of the reviewer. Thus we make it harder to skip it during the review. We don't want to treat it as a pattern. On contrary, we want to keep its usage to minimum.

I will unfold it back.